### PR TITLE
Introduciton and Domain analysis section

### PR DESCRIPTION
**Pull request by HenrikSkindhoj**

### CODE REVIEW

**Review by Kageschwump — APPROVED** [review]

**Comment by muItilingualism** [comment]
Merge into dev instead of main pls

File: report/introduction.md [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
-#### Project Scope and Objectives
+#### 1.1 Project Scope and Objectives
 
-#### Problem Statement and Solution Overview
+The goal of this project is to design and implement a full-stack gambling platform, with games such as roulette, and coin flip. 
+The project includes the development of both a backend, a frontend, and the use of DevOps practices to improve the development and deployment processes.
 
-#### Methodology
+The project will provide an secure and intuitive user experience. Software using modern frameworks and tools will be implemented. 
+The project will be implemented with an emphasis on using the practices of DevOps to enable continous integration and deployment.
```
Comment (muItilingualism) [inline report/introduction.md]: There is nothing about the scope in this

File: report/introduction.md [hunk 2]
```diff
@@ -1,8 +1,51 @@
...
+
+##### Problem Statement:
+
+In this project several challenges had to be overcome, these include:
+
+- **Scalability:** Making it possible for several different users to play simultaneously, while maintaining uninterrupted gameplay.
+- **Security:** Preventing exploitation of tokens, to prevent potential threats.
+- **Maintainability:** Creating a system that can be updated and improvied without any significant overhead.
+- **Deployment:** Developing a deployment process with frequent updates without downtime
+
```
Comment (muItilingualism) [inline report/introduction.md:19]: Explain why we chose these

File: report/introduction.md [hunk 3]
```diff
@@ -1,8 +1,51 @@
...
+  - **Containerization** with Docker to provide environments that are consistent across development, testing, and production.
+  - **Version control and automation** using GitHub and GitHub Actions for continuous integration and deployment.
+- An emphasis on **secure authentication and authorization**, by using **JWT tokens** to manage user sessions effectively.
+
+These approaches both addresses the technical challenges, and ensures that the platform is secure, and scalable.
+
+
+#### 1.3 Methodology
+
+The project follows the Agile methodology of iterative development in a manner that allows for easier addaptation to changes in requirements and the users needs.
```
Comment (muItilingualism) [inline report/introduction.md]: when did we use agile?

Comment (muItilingualism) [reply report/introduction.md]: Also "addaptation" misspelling

File: report/introduction.md [hunk 4]
```diff
@@ -1,8 +1,51 @@
...
+
+
+#### 1.3 Methodology
+
+The project follows the Agile methodology of iterative development in a manner that allows for easier addaptation to changes in requirements and the users needs. 
+The process initiates with comprehensive requirements gathering and analysis, where user stories and use cases are written in detail to ensure alignment with end-user expectations. 
+This approach promotes clarity of purpose for the project and lays a solid foundation for later phases in development.
+
+The development phase involves the creation of both the frontend and backend, each designed with scalability, and maintainability in mind. 
+The structure of the frontend is component-based, built with React and TypeScript, promoting reusability. This makes it easier for developers to update and extend the user interface every time something a new feature is implemented.
```
Comment (muItilingualism) [inline report/introduction.md]: This should not be in the methodology.

File: report/introduction.md [hunk 5]
```diff
@@ -1,8 +1,51 @@
...
+#### 1.3 Methodology
+
+The project follows the Agile methodology of iterative development in a manner that allows for easier addaptation to changes in requirements and the users needs. 
+The process initiates with comprehensive requirements gathering and analysis, where user stories and use cases are written in detail to ensure alignment with end-user expectations. 
+This approach promotes clarity of purpose for the project and lays a solid foundation for later phases in development.
+
+The development phase involves the creation of both the frontend and backend, each designed with scalability, and maintainability in mind. 
+The structure of the frontend is component-based, built with React and TypeScript, promoting reusability. This makes it easier for developers to update and extend the user interface every time something a new feature is implemented. 
+The backend uses a layered architecture based on Quarkus, which isolates concerns into three layers for business logic, API endpoints, and data management. 
+This makes the code cleaner and easier to scale, making it easier to integrate further features.
```
Comment (muItilingualism) [inline report/introduction.md]: This, also, should not be in the methodology

File: report/introduction.md [hunk 6]
```diff
@@ -1,8 +1,51 @@
...
+
+DevOps practices are integrated into the project to improve the development and deployment processes. Continuous integration and deployment pipelines will be implemented with the help of GitHub Actions where code builds, testing, and deployments are automated. It minimizes manual labour, reduces the possibilities of human errors, and speeds up the release cycle. Docker is used for containerization, which maintains the consistency of the environments throughout the development, testing, and production stages, enhancing reliability and simplifying the deployment process.
```
Comment (muItilingualism) [inline report/introduction.md]: This should also not be in the methodology

File: report/introduction.md [hunk 7]
```diff
@@ -1,8 +1,51 @@
...
+
+The deployment strategy adopted here is cloud-based, with the backend deployed to Google Cloud, and the frontend hosted on Netlify.
+
+By integrating these methodologies, the project ensures a smooth development process, that's responsive to change.
```
Comment (muItilingualism) [inline report/introduction.md]: The methodology segment should not describe the technology used, it should describe the methods used.

Please instead, write/describe 

1. development lifecycle model of the project
2. our methods for gathering requirements and analyzing the problem space
3. project management approach

**Review by muItilingualism — CHANGES_REQUESTED** [review]

**Review by muItilingualism — APPROVED** [review]

**Pull request merged**